components/kyma-environment-broker/internal/ias/assertion_attribute.go

```diff
@@ -0,0 +1,63 @@
+package ias
+
+// AssertionAttributeDeliver ensures required AssertionAttributes
+// instead remove all and replace by new one, it will remove only not existing in templates
+// and leave existing with probably fresher version of user attributes
+type AssertionAttributeDeliver struct {
+	assertionAttributesTemplate map[string]AssertionAttribute
+}
+
+// NewAssertionAttributeDeliver returns new AssertionAttributeDeliver with default attributes template
+func NewAssertionAttributeDeliver() *AssertionAttributeDeliver {
+	return &AssertionAttributeDeliver{
+		assertionAttributesTemplate: map[string]AssertionAttribute{
+			"first_name": {
+				AssertionAttribute: "first_name",
+				UserAttribute:      "firstName",
+			},
+			"last_name": {
+				AssertionAttribute: "last_name",
+				UserAttribute:      "lastName",
+			},
+			"email": {
+				AssertionAttribute: "email",
+				UserAttribute:      "mail",
+			},
+			"groups": {
+				AssertionAttribute: "groups",
+				UserAttribute:      "companyGroups",
+			},
+		},
+	}
+}
+
+func (a *AssertionAttributeDeliver) assertionAttributesKeys() map[string]bool {
+	attributes := make(map[string]bool, len(a.assertionAttributesTemplate))
+	for key, _ := range a.assertionAttributesTemplate {
+		attributes[key] = false
+	}
+
+	return attributes
+}
+
+// GenerateAssertionAttribute remove not existing in template attributes, leaves existing
+func (a *AssertionAttributeDeliver) GenerateAssertionAttribute(serviceProvider ServiceProvider) []AssertionAttribute {
+	var newAssertionAttributes []AssertionAttribute
+	aats := a.assertionAttributesKeys()
+
+	for key, _ := range aats {
```

Comment:
`for in for` always scares me. Also, this map with bools was problematic to understand at the beginning
WDYT about this? IMO is shorter and more readable

```go
	defaults := a.tmplDeepCopy()
	// overrides defaults with given attr
	for _, overrideAtr := range serviceProvider.AssertionAttributes {
		if _, found := defaults[overrideAtr.AssertionAttribute]; found {
			defaults[overrideAtr.AssertionAttribute] = overrideAtr
		}
	}

	// convert to slice
	var newAssertionAttributes []AssertionAttribute
	for _, atr := range defaults {
		newAssertionAttributes = append(newAssertionAttributes, atr)
	}
```

just replace `assertionAttributesKeys` function with:
```go
func (a AssertionAttributeDeliver) tmplDeepCopy() map[string]AssertionAttribute {
	cpy := map[string]AssertionAttribute{}
	for k, v := range a.assertionAttributesTemplate {
		cpy[k] = v
	}
	return cpy
}
```

Reply:
done.